lib/index.d.ts

```diff
@@ -15,12 +15,49 @@ export * from "./joint";
 export * from "./shape";
 export * from "../testbed";
 
-export type Manifold = any; // TODO
+export const enum ContactFeatureType {
+  Vertex = 0,
+  Face = 1
+}
+
+export interface ContactFeature {
+  indexA: number;
+  indexB: number;
+  typeA: ContactFeatureType;
+  typeB: ContactFeatureType;
+}
 
-export type WorldManifold = any; // TODO
+export interface ContactID {
+  cf: ContactFeature;
```

Comment:
Maybe we could also add the `key` property as readonly. I guess the `set` method of `ContactID` and `ContactFeature` doesn't have to be exposed?
@shakiba Do you think `key` or `set` is used in the wild?

Reply:
Indeed I did not check the readonly-ness of properties, I can have a look. Any input is welcome.

Reply:
@zOadT you mean putting all properties of both `ContactID` and `ContactFeature` as `readonly`, right?

Reply:
I was just talking about the property `key` of `ContactID` (it's defined in line 113 of Manifold.js).
You are probably right that it makes sence to set `CantactID` immutable, but I think we don't need to enforce it in the types as it is practically doable to change these values. But let me know what you think.